13-1-1/update_indicator.R

```diff
@@ -0,0 +1,387 @@
+  }
+  combinations <- combinations %>% 
+    mutate(Value = placeholder)
+  return(combinations)
+}
+
+# This is for nomis_deaths. Aggregates Value for the 10 causes of death
+get_cod_aggregates <- function(df){
+  combinations <- unique(df[c("Country", "Year", "Sex")])
+  placeholder <- vector("integer", nrow(combinations))
+  for(i in 1:nrow(combinations)){
+    cty<-combinations[i,]$Country 
+    yr<-combinations[i,]$Year 
+    sx<-combinations[i,]$Sex
+    toAggregate <- df %>% filter(Country==cty, Year==yr, Sex==sx) %>% select(Value)
+    placeholder[i] <- colSums(toAggregate)
+  }  
+  combinations <- combinations %>% 
+    mutate(Value = placeholder)
+  return(combinations)
+}
+
+# Clean up the ons_deaths tables -----
+# ons_deaths1 is all COD all sex (no MALE AND FEMALE aggregate supplied)
+header_ons_1 <- which(ons_deaths1$X1 == "Country")
+names(ons_deaths1) <- ons_deaths1[header_ons_1,]
+ons_deaths1 <- ons_deaths1 %>% 
+  tail(-header_ons_1)
+
+ons_deaths1 <- ons_deaths1 %>% 
+  select(Country, `Registration year`, Deaths) %>% 
+  tidyr::fill(c('Country')) %>% 
+  rename(c(Year=`Registration year`, Value=Deaths)) %>% 
+  mutate(Value = as.numeric(Value)) %>% 
+  mutate(Series = "Number of deaths from exposure to forces of nature") %>% 
+  mutate(Sex = "") %>% 
+  mutate(Country = ifelse(Country=="England and wales", "", Country)) %>% 
+  mutate(`Observation status` = "Normal value") %>% 
+  mutate(Units = "Number") %>% 
+  mutate(`Unit multiplier` = "") %>% 
+  mutate(`Cause of death` = "") %>% 
+  mutate(GeoCode = lapply(Country, get_GeoCode)) %>% 
+  mutate(GeoCode = as.character(GeoCode)) %>% 
+  select(Year, Series, `Cause of death`, Country, Sex, `Observation status`, `Unit multiplier`, `Units`, GeoCode, Value)
+
+# ons_deaths2 is all COD. Need to add rows for the sex aggregates
+header_ons_2 <- which(ons_deaths2$X1 == "Country")
+names(ons_deaths2) <- ons_deaths2[header_ons_2,]
+ons_deaths2 <- ons_deaths2 %>% 
+  tail(-header_ons_2)
+
+ons_deaths2 <- ons_deaths2 %>% 
+  select(Country, Sex, `Registration year`, Deaths) %>% 
+  tidyr::fill(c('Country', 'Sex')) %>% 
+  rename(c(Year=`Registration year`, Value=Deaths)) %>% 
+  mutate(Value = as.numeric(Value)) %>% 
+  mutate(Country = ifelse(Country=="England and wales", "", Country)) %>% 
+  mutate(Series = "Number of deaths from exposure to forces of nature") %>% 
+  mutate(`Observation status` = "Normal value") %>% 
+  mutate(Units = "Number") %>% 
+  mutate(`Unit multiplier` = "") %>% 
+  mutate(`Cause of death` = "") %>% 
+  mutate(GeoCode = lapply(Country, get_GeoCode)) %>% 
+  mutate(GeoCode = as.character(GeoCode)) %>% 
+  select(Year, Series, `Cause of death`, Country, Sex, `Observation status`, `Unit multiplier`, `Units`, GeoCode, Value)
+
+# Now we need to add rows for the aggregates of Sex which aren't
+# present in the source data
+
+ons_death2_sex_agg <- get_sex_aggregates_v2(ons_deaths2)
+
+ons_death2_sex_agg <- ons_death2_sex_agg %>% 
```

Comment:
I'm pretty sure you can delete this (see above comment. 
Also, just to note, ot makes your life harder in the long run to name a df the same as another df as you do here - it makes running bits of code out of order both dangerous and annoying. Go with meaningful names. e.g. here, (if you were keeping it) you could have `sex_agg_cleaned <- sex_agg %>% ...`